src/java/htsjdk/samtools/SAMRecordDuplicateComparator.java

```diff
@@ -39,6 +39,10 @@
  * @author nhomer
  */
 public class SAMRecordDuplicateComparator implements SAMRecordComparator {
+    /** An enum to provide type-safe keys for transient attributes the comparator puts on SAMRecords. */
+    private static enum Attr {
+        LibraryId, ReadCoordinate, MateCoordinate
```

Comment:
I really like the approach I've taken where the using class declares a private enum and uses the values as the key.  It essentially makes key collision impossible and hides transient attributes from classes that didn't create them/don't have access to the enum.  So I'd prefer not to centralize.